server/graphql/v2/input/TierReferenceInput.js

```diff
@@ -1,11 +1,38 @@
-import { GraphQLInputObjectType, GraphQLInt } from 'graphql';
+import { GraphQLInputObjectType, GraphQLString } from 'graphql';
+
+import models from '../../../models';
+import { NotFound } from '../../errors';
+import { idDecode } from '../identifiers';
 
 export const TierReferenceInput = new GraphQLInputObjectType({
   name: 'TierReferenceInput',
   fields: () => ({
-    legacyId: {
-      type: GraphQLInt,
-      description: 'The legacy id assigned to the Tier',
+    id: {
+      type: GraphQLString,
+      description: 'The id assigned to the Tier',
     },
   }),
 });
+
+/**
+ * Retrieves a tier
+ *
+ * @param {string|number} input - id of the tier
+ */
+export const fetchTierWithReference = async input => {
+  const loadTierById = id => {
```

Comment:
Looks like it's not needed, you can simply use `models.Tier.findByPk` later.

Reply:
Why would we use `models.Tier.findByPk` in this case but `fetch(Account/PaymentMethod)WithReference` in other places?

Reply:
I'm writing about `loadTierById` which looks like an useless abstraction.